pkg/controller/control.go

```diff
@@ -660,6 +697,12 @@ func (c *Controller) Start(addresses ...string) error {
 	// For more details, see the following url:
 	// https://github.com/longhorn/longhorn/issues/1235
 	if !c.isUpgrade {
+		// Check if the replicas are in valid state.
+		// If not in a valid state, mark the replica as ERR.
+		if err := c.checkReplicasState(); err != nil {
+			return err
+		}
```

Comment:
I think we can move this as [a part of function `c.Start()`](https://github.com/longhorn/longhorn-engine/blob/7310e09cd6632276f29ae8659a1388b9ee8edbc7/pkg/controller/control.go#L605-L649). 
Then you can eliminate some duplications for `checkReplicasState`

Reply:
Updated. Thank you.